indexer/src/main/java/bio/terra/tanagra/indexing/job/dataflow/WriteInstanceLevelDisplayHints.java

```diff
@@ -134,17 +145,32 @@ public void run(boolean isDryRun) {
         readInRelationshipIdPairs(
             pipeline, occPriIdPairsSql, entityAIdColumnName, entityBIdColumnName);
 
+    PCollection<KV<Long, Long>> rollupOccCriIdPairKVs = null;
+    if (hierarchy != null
+        && criteriaOccurrence.hasRollupInstanceLevelDisplayHints(occurrenceEntity)) {
+      PCollection<KV<Long, Long>> ancestorDescendantRelationshipsPC =
+          BigQueryBeamUtils.readAncestorDescendentRelationshipsFromBQ(
+              pipeline, ancestorDescendantTable);
+
+      // Expand the set of occurrences to include a repeat for each ancestor.
+      rollupOccCriIdPairKVs =
+          CountUtils.repeatOccurrencesForHints(occCriIdPairKVs, ancestorDescendantRelationshipsPC);
+    }
+    final PCollection<KV<Long, Long>> finalRollupOccCriIdPairKVs = rollupOccCriIdPairKVs;
+
     criteriaOccurrence
         .getAttributesWithInstanceLevelDisplayHints(occurrenceEntity)
         .forEach(
-            attribute -> {
+            (attribute, rollup) -> {
+              PCollection<KV<Long, Long>> idPairsKVs =
+                  rollup ? finalRollupOccCriIdPairKVs : occCriIdPairKVs;
               if (attribute.isValueDisplay()) {
                 LOGGER.info("enum val hint: {}", attribute.getName());
-                enumValHint(occCriIdPairKVs, occPriIdPairKVs, occIdRowKVs, attribute);
+                enumValHint(idPairsKVs, occPriIdPairKVs, occIdRowKVs, attribute);
```

Comment:
enumValHint and numericRangeHint do not handle null values. 

line 148: 
final rollupOccCriIdPairKVs == empty collection
line 156: add all
line 159 could be removed. 

alternately - line 159: assign empty collection if null

Reply:
Thats fine, as long as its intentional behavior